### PR TITLE
Default to delambdafy:method and backend:GenBCode

### PR DESCRIPTION
**Pull request by lrytz**
Swtich the defaults of `-Ydelambdafy` and `-Ybackend`.

### CODE REVIEW

**Comment by lrytz** [comment]
failures expected, will require a rebase once #4578 gets merged.

File: src/compiler/scala/tools/nsc/settings/ScalaSettings.scala [hunk 1]
```diff
@@ -224,7 +224,7 @@ trait ScalaSettings extends AbsScalaSettings
 
   val exposeEmptyPackage = BooleanSetting("-Yexpose-empty-package", "Internal only: expose the empty package.").internalOnly()
   // the current standard is "inline" but we are moving towards "method"
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/settings/ScalaSettings.scala]: Comment is out of date now, I'd just remove it.

**Comment by lrytz** [comment]
```
Test suite finished with 19 cases failing
```

looking into those today

**Comment by lrytz** [comment]
@retronym there's a number of tests that need to be rewritten, for example those using `IcodeComparison` (no more icode). In the interest of not getting blocked on this PR, I propose to move those that need rewriting to pending. wdyt? I'll continue working on them today / tomorrow.

**Comment by retronym** [comment]
That's fine with me. Let's remind ourselves with a blocker ticket for m3

**Comment by lrytz** [comment]
@retronym fixed the tests. The commit here is a squashed version of this: https://github.com/scala/scala/compare/2.12.x...lrytz:defaultsWip?expand=1

**Comment by retronym** [comment]
LGTM

**Pull request merged**